daml-script/daml/Daml/Script.daml

```diff
@@ -226,32 +226,32 @@ data QueryContractIdPayload a = QueryContractIdPayload
 -- Returns `None` if there is no active contract the party is a stakeholder on.
 -- This is semantically equivalent to calling `query`
 -- and filtering on the client side.
-queryContractId : forall t p. (Template t, IsParties p) => HasCallStack => p -> ContractId t -> Script (Optional t)
+queryContractId : forall t p. (Template t, IsParties p) => HasCallStack => p -> ContractId t -> Script (Optional (t, ContractMetadata))
 queryContractId p c = lift $ Free $ QueryContractId QueryContractIdPayload with
   parties = toParties p
   tplId = templateTypeRep @t
   cid = coerceContractId c
-  continue = pure . fmap (fromSome . fromAnyTemplate)
+  continue = pure . fmap (\(anyTpl, meta) -> (fromSome (fromAnyTemplate anyTpl), meta))
   locations = getCallStack callStack
 
 data QueryContractKeyPayload a = QueryContractKeyPayload
   { parties : [Party]
   , tplId : TemplateTypeRep
   , key : AnyContractKey
-  , continue : Optional (ContractId (), AnyTemplate) -> a
+  , continue : Optional (ContractId (), AnyTemplate, ContractMetadata) -> a
```

Comment:
it seems inconsistent in when you use a 3-tuple and when you use nested 2-tuples.